### ISSUE TITLE
[Bug] Neoforge 1.20.1 Freeze on joining world

### ISSUE THREAD

**Issue by HoodBlah** [TimVincii/Terrastorage#8]
### What happened?

When joining a world on Neoforge 1.20.1 the game freeze on the loading screen. Upon looking at the initial logs is seemed that Zeta was incompatible but the issue persisted even after removing Zeta and its dependencies. 

I then tried with just Terrastorage and Integrated Stronghold (+dependencies) on a fresh install and got the same issue.

I did further testing and narrowed down the list to be Integrated API, Integrated Stronghold, Zeta, and Quark. One or all of these mods are causing the crash. The other mods; Create, Moonlight Library, and Supplementary worked okay.

----------------------
First time with Zeta
[debug(with zeta).log](https://github.com/user-attachments/files/17549871/debug.with.zeta.log)
[log (with zeta).log](https://github.com/user-attachments/files/17549872/log.with.zeta.log)

Second time without Zeta
[debug.log](https://github.com/user-attachments/files/17549874/debug.log)
[latest.log](https://github.com/user-attachments/files/17549875/latest.log)

Fresh install with Terrastorage and Integrated Stronghold (+dependencies)
[debug.log](https://github.com/user-attachments/files/17549951/debug.log)
[latest.log](https://github.com/user-attachments/files/17549952/latest.log)

Sorry to throw another crash at you right after you fixed the last one bro 😅

### Terrastorage Version

1.1.2

### Fabric API Version

Forgified Fabric API 0.92.2+1.11.8+1.20.1

### Minecraft Version

1.20.1

### Other Mods

```shell
Create 0.5.1.j
Forgified Fabric API 0.92.2+1.11.8+1.20.1
Sinytra Connector Connector-1.0.0-beta.46+1.20.1
Integrated API integrated_api-1.5.1+1.20.1-forge
Integrated Stronghold integrated_stronghold-1.1.1+1.20.1-forge
Moonlight Library moonlight-1.20-2.13.18-forge
Quark-4.0-460
supplementaries-1.20-2.8.17
Terrastorage-1.1.2+1.20-1.20.1
Zeta-1.0-24
```

**Comment by TimVincii** [comment]
Hey again, no need to be sorry! if anything, I'm sorry that Terrastorage is so unusable on NeoForge, back when I just started making it, I didn't think at all about using a mutliloader template to provide support for loaders other than Fabric and Quilt, and I kinda regret that now.

Thanks a lot for your thorough bug report and for testing with different mod setup, that helped me narrow down the issue quickly.

Seems like Quark or Zeta were having issues with the fact that Terrastorage was trying to execute some logic while the server was starting (even in Singleplayer, a server is technically running within the client).
I was able to fix this by moving that logic to execute after the server finishes starting up. This fix is now live in Terrastorage 1.1.3, so worlds should load correctly with that update:
* [[Modrinth] Terrastorage 1.1.3+1.20-1.20.1](https://modrinth.com/mod/terrastorage/version/1.1.3+1.20-1.20.1)
* [[CurseForge] Terrastorage 1.1.3+1.20-1.20.1](https://www.curseforge.com/minecraft/mc-mods/terrastorage/files/5857914)

While testing, I found a separate bug: nametags of renamed chests and shulker boxes aren’t rendering properly when Terrastorage is used with Create on NeoForge. I’ll be looking into that next.

Thank you again for your time and dedication in reporting this and the previous bug. I'll now be closing this issue as completed, and I hope you have a great day!
